source/Ship.cpp

```diff
@@ -2671,7 +2703,8 @@ int Ship::Scan()
 			Messages::Add("Attempting to scan the selected " + target->Noun() + "."
 				, Messages::Importance::Low);
 
-		if(target->GetPersonality().IsSecretive() && target->GetGovernment()->IsProvokedOnScan())
+		if(target->GetPersonality().IsSecretive() && target->GetGovernment()->IsProvokedOnScan()
+			&& target->CanSendHail(player))
```

Comment:
Shouldn't this be:
```suggestion
		if((target->GetPersonality().IsSecretive() || target->GetGovernment()->IsProvokedOnScan())
			&& target->CanSendHail(player))
```
They shouldn't have to be *both* secretive and provoked on scan to give a warning, should they?
Or, even, secretive ships just shouldn't send this message at all? If you're trying to keep something a secret, you may not necessarily wish to advertise it.

Reply:
I think it works best with both tbh
Not all provoked on ships would say smth (also not the idea of this PR to change this behavior)

That said, if thats the only blocker of the PR do feel free to change it

Reply:
Smugglers for instance would be secretive but would not say anything because not provoked on scan

Reply:
```suggestion
		if(target->GetGovernment()->IsProvokedOnScan() && target->CanSendHail(player))
```
This is acceptable

Reply:
Requiring both means that something that is provoked on scan but isn't secretive wouldn't give the warning hails.

Reply:
Alr then, only use case for provoked on scan is the remnants anyway and it works for them
Lets just do this and change it later if needed